### PR TITLE
[prometheus-kafka-exporter] - added SASL support and upgrade to latest version

### PR DESCRIPTION
**Pull request by maxpschonder**
Signed-off-by: Max Pschonder <max.pschonder@hivemindtechnologies.com>

#### What this PR does / why we need it:
I added support for SASL configuration options and fixed errors caused by unsupported `—log.level` param.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - SASL support
  - `--verbosity` replaced `--log-level` [README Update](https://github.com/danielqsj/kafka_exporter/commit/b6cb6afe3bc77ff0a7ce556f33f9397e57f9b63b)

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by maxpschonder** [comment]
Hey @golgoth31, maybe you could also take a look at it, please?

File: charts/prometheus-kafka-exporter/Chart.yaml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 apiVersion: v2
-appVersion: "v1.3.1"
+appVersion: "v1.3.2"
 description: A Helm chart to export the metrics from Kafka in Prometheus format using the kafka-exporter from https://github.com/danielqsj/kafka_exporter
 name: prometheus-kafka-exporter
 home: https://github.com/danielqsj/kafka_exporter
-version: 1.4.0
+version: 1.4.1
```
Comment (golgoth31) [inline charts/prometheus-kafka-exporter/Chart.yaml:6]: Adding such functionality, we should go to 1.5.0, I think

File: charts/prometheus-kafka-exporter/Chart.yaml [hunk 2]
```diff
@@ -1,9 +1,9 @@
 apiVersion: v2
-appVersion: "v1.3.1"
+appVersion: "v1.3.2"
```
Comment (golgoth31) [inline charts/prometheus-kafka-exporter/Chart.yaml:2]: appVersion should be 1.4.1 as set in values.yaml

File: charts/prometheus-kafka-exporter/values.yaml [hunk 3]
```diff
@@ -103,6 +105,23 @@ tls:
   # mountPath: /secret/certs
   # secretName: <name of an existing secret>
 
+sasl:
+  enabled: false
+  handshake: true
+  scram:
+    enabled: false
```
Comment (golgoth31) [inline charts/prometheus-kafka-exporter/values.yaml:112]: Can you add a way to declare an existing secret for scram ?

**Review by golgoth31 — COMMENTED** [review]
Thank you for this !

File: charts/prometheus-kafka-exporter/Chart.yaml (hunk 1 above)
Comment (maxpschonder) [reply charts/prometheus-kafka-exporter/Chart.yaml]: i agree

File: charts/prometheus-kafka-exporter/Chart.yaml (hunk 2 above)
Comment (maxpschonder) [reply charts/prometheus-kafka-exporter/Chart.yaml]: of course, will do.

File: charts/prometheus-kafka-exporter/values.yaml (hunk 3 above)
Comment (maxpschonder) [reply charts/prometheus-kafka-exporter/values.yaml:112]: done. can you think of a more concise way?

Comment (golgoth31) [reply charts/prometheus-kafka-exporter/values.yaml:112]: it's not shorter but you can use a template in the _helpers.yaml to write the if statement only once

File: charts/prometheus-kafka-exporter/templates/secret.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{{- if .Values.sasl.scram.enabled }}
```
Comment (golgoth31) [inline charts/prometheus-kafka-exporter/templates/secret.yaml:1]: Can disable this secret creation if using an existing secret ?

**Review by golgoth31 — APPROVED** [review]

**Comment by golgoth31** [comment]
From my point of view, all is ok. Thank you for the job !

**Review by golgoth31 — APPROVED** [review]

**Comment by maxpschonder** [comment]
Thank you for the review and inputs! 🙇‍♂️

**Comment by golgoth31** [comment]
@gkarthiks Are you ok to validate this PR ?

**Review by gkarthiks — APPROVED** [review]

**Comment by gkarthiks** [comment]
It looks good, but I am actualy good with your approval @golgoth31.

**Pull request merged**